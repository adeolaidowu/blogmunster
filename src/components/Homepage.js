import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";
import PostList from "./PostList";
import PostFilters from "./PostFilters";

export const Homepage = ({ startLogin }) => (
  <div>
    <h1>Welcome to BlogMunster</h1>
    <button onClick={startLogin}>Login with Google</button>
    {<PostFilters />}
    {<PostList />}
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(null, mapDispatchToProps)(Homepage);
