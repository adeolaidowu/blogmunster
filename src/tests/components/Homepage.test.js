import React from "react";
import Homepage from "../../components/Homepage";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

test("should render homepage correctly", () => {
  const wrapper = shallow(<Homepage />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
