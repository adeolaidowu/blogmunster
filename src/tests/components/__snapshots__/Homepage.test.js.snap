// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render homepage correctly 1`] = `
<div>
  <h1>
    Welcome to BlogMunster
  </h1>
  <button
    onClick={[Function]}
  >
    Login with Google
  </button>
  <Connect(PostFilters) />
  <Connect(PostList) />
</div>
`;
